packages/design-system-react/src/components/button-base/ButtonBase.stories.tsx

```diff
@@ -0,0 +1,177 @@
+import type { Meta, StoryObj } from '@storybook/react';
+import React from 'react';
+
+import { IconName } from '..';
+import { ButtonBase } from './ButtonBase';
+import { ButtonBaseSize } from './ButtonBase.types';
+import README from './README.mdx';
+
+const meta: Meta<typeof ButtonBase> = {
+  title: 'React Components/ButtonBase',
+  component: ButtonBase,
+  parameters: {
+    docs: {
+      page: README,
+    },
+  },
+  argTypes: {
+    children: {
+      control: 'text',
+      description:
+        'Required prop for the content to be rendered within the ButtonBase',
+    },
+    className: {
+      control: 'text',
+      description:
+        'Optional prop for additional CSS classes to be applied to the ButtonBase component',
+    },
+    size: {
+      control: 'select',
+      options: Object.values(ButtonBaseSize),
+      description: 'Optional prop to control the size of the ButtonBase',
+      table: {
+        defaultValue: { summary: 'ButtonBaseSize.Md' },
+      },
+    },
+    isFullWidth: {
+      control: 'boolean',
+      description:
+        'Optional prop that when true, makes the button take up the full width of its container',
+    },
+    asChild: {
+      control: 'boolean',
+      description:
+        'Optional boolean that determines if the component should merge its props onto its immediate child instead of rendering a button element',
+    },
+    isLoading: {
+      control: 'boolean',
+      description: 'Optional prop that when true, shows a loading spinner',
+    },
+    loadingText: {
+      control: 'text',
+      description:
+        'Optional prop for text to display when button is in loading state',
+    },
+    startIconName: {
+      control: 'select',
+      options: Object.values(IconName),
+      description:
+        'Optional prop to specify an icon to show at the start of the button',
+    },
+    startIconProps: {
+      control: 'object',
+      description:
+        'Optional prop to pass additional properties to the start icon',
+    },
+    startAccessory: {
+      control: 'text',
+      description:
+        'Optional prop for a custom element to show at the start of the button',
+    },
+    endIconName: {
+      control: 'select',
+      options: Object.values(IconName),
+      description:
+        'Optional prop to specify an icon to show at the end of the button',
+    },
+    endIconProps: {
+      control: 'object',
+      description:
+        'Optional prop to pass additional properties to the end icon',
+    },
+    endAccessory: {
+      control: 'text',
+      description:
+        'Optional prop for a custom element to show at the end of the button',
+    },
+    isDisabled: {
+      control: 'boolean',
+      description: 'Optional prop that when true, disables the button',
+    },
+  },
```

Comment:
Adds controls to story and improves component API section

<img width="1511" alt="Screenshot 2024-12-10 at 4 31 24 PM" src="https://github.com/user-attachments/assets/97716de8-d056-4ed0-bee3-fa301693b537">
<img width="1479" alt="Screenshot 2024-12-10 at 4 31 40 PM" src="https://github.com/user-attachments/assets/a9902ce9-7117-40ef-8c8c-b8087a82b9ff">